packages/dev/docs/pages/react-spectrum/testing.mdx

```diff
@@ -156,3 +158,198 @@ userEvent.type(document.activeElement, 'Pas$w0rd');
 userEvent.tab();
 userEvent.click(document.activeElement);
 ```
+
+## Test setup and common gotchas
+
+### Providers
+
+When testing anything that uses a React Spectrum component, you must provide a wrapping Provider at the top level. Components rely on this wrapping provider
+to inform them of the app's theme, locale, breakpoints and more. See the [Provider page](./Provider.html#application-provider) for a list of properties
+your Provider can accept. If you see the following errors, you should check to see if you have a wrapping Provider.
+
+```
+Error: Uncaught [TypeError: Cannot destructure property 'scale' of '(0 , _provider.useProvider)(...)' as it is null.]
+```
+
+```
+TypeError: Cannot read properties of null (reading 'scale')
+```
+
+If you are using React Testing Library, you can overwrite the provided `render` function so that it automatically includes a Provider by default.
+Note the below example allows you to rerender the component with new Provider props but you'll need to requery any element you are asserting against
+to get its updated state (e.g. `button = getByRole('button)`).
+
+```tsx
+import {RenderOptions as BaseRenderOptions, render} from '@testing-library/react';
+import {Provider, defaultTheme} from '@adobe/react-spectrum';
+import React, {ReactElement} from 'react';
+
+// export everything from React Testing Library
+export * from '@testing-library/react';
+
+function ProviderWrapper(props) {
+  let {children, ...providerProps} = props;
+  return (
+		<Provider theme={defaultTheme} scale="medium" {...providerProps}>
```

Comment:
so much indent

Reply:
oh thats weird, I don't see that locally in my IDE :/, lemme try to fix